File: test-suite-docs-python/src/test/python/micronaut/mcp/docs/completions/resources/MyResourceCompletions.py
# tag::imports[]
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.mcp.annotations import ResourceCompletion
# end::imports[]


@Requires(property="spec.name", value="MyResourceCompletionsTest")
# tag::clazz[]
@Singleton
class MyResourceCompletions:
    @ResourceCompletion(uri="file:///home/user/documents/{fileName}")
    def resources_completions(self, fileName: str) -> list[str]:
        return [name for name in ["report.pdf", "data.csv", "notes.txt"] if name.startswith(fileName)]
# end::clazz[]

File: test-suite-docs-java/src/test/java/io/micronaut/mcp/docs/completions/resources/MyResourceCompletions.java
package io.micronaut.mcp.docs.completions.resources;

//tag::imports[]
import io.micronaut.context.annotation.Requires;
import io.micronaut.mcp.annotations.ResourceCompletion;
import jakarta.inject.Singleton;

import java.util.List;

//end::imports[]
@Requires(property = "spec.name", value = "MyResourceCompletionsTest")
//tag::clazz[]
@Singleton
class MyResourceCompletions {
    @ResourceCompletion(uri = "file:///home/user/documents/{fileName}")
    List<String> resourcesCompletions(String fileName) {
        return List.of(
                "report.pdf",
                "data.csv",
                "notes.txt"
            ).stream()
            .filter(name -> name.startsWith(fileName))
            .toList();
    }
}
//end::clazz[]
